File: DSA/SakshamDwivedi_590013147/Day18/D18P1.py
#Selection Sort
def selection_sort(arr):
    for i in range(len(arr) - 1):
        min_ind = i
        for j in range(i+1, len(arr)):
            if(arr[min_ind] > arr[j]):
                min_ind = j
        arr[i], arr[min_ind] = arr[min_ind], arr[i]
    return arr

#Fixed Input
print(selection_sort([3,2,6,1,7]))

#Custom Input
arr = list(int(x) for x in input("Enter the array : ").split())
print(selection_sort(arr))

File: DSA/SakshamDwivedi_590013147/Day18/D18P1.c
#include <stdio.h>

int arr[100];
int size;
int min_ind;

int main(){
    printf("Enter the size of the array (Less than 100) : ");
    scanf("%d", &size);
    printf("Enter the elements : ");
    for(int i = 0; i < size; i++){
        scanf("%d", &arr[i]);
    }

    for(int i = 0; i < size; i++){
        min_ind = i;
        for(int j = i+1; j < size; j++){
            if(arr[min_ind] > arr[j]){
                min_ind = j;
            }
        }
        int temp = arr[i];
        arr[i] = arr[min_ind];
        arr[min_ind] = temp;
    }

    printf("Sorted Array : ");
    for(int i = 0; i < size; i++){
        printf("%d ", arr[i]);
    }

}
